tests/test_benchmark.py

```diff
@@ -230,3 +240,262 @@ def test_no_stock_dividends_allowed(self):
+                                context.has_ordered = True
+                        """)
+
+    @classmethod
+    def make_equity_info(cls):
+        cls.set_expected_outcome()
+        return pd.DataFrame.from_dict(
+            {
+                1: {
+                    'symbol': 'A',
+                    'start_date': cls.START_DATE,
+                    'end_date': cls.END_DATE + pd.Timedelta(days=1),
+                    "exchange": "TEST",
+                },
+                2: {
+                    'symbol': 'B',
+                    'start_date': cls.START_DATE,
+                    'end_date': cls.END_DATE + pd.Timedelta(days=1),
+                    "exchange": "TEST",
+                }
+            },
+            orient='index',
+        )
+
+    @classmethod
+    def make_equity_daily_bar_data(cls, country_code, sids):
+        yield 1, pd.DataFrame(
+            {
+                'open': [100, 120, 100, 160, 180, 200],
+                'high': [100, 120, 100, 160, 180, 200],
+                'low': [100, 120, 100, 160, 180, 200],
+                'close': [100, 120, 100, 160, 180, 200],
+                'volume': 100,
+            },
+            index=cls.equity_daily_bar_days,
+        )
+
+        yield 2, pd.DataFrame(
+            {
+                'open': [100, 90, 120, 140, 160, 180],
+                'high': [100, 90, 120, 140, 160, 180],
+                'low': [100, 90, 120, 140, 160, 180],
+                'close': [100, 90, 120, 140, 160, 180],
+                'volume': 100,
+            },
+            index=cls.equity_daily_bar_days,
+        )
+
+    @classmethod
+    def set_expected_outcome(cls):
+        cls.expected_daily = {
+            'returns': np.array([0., 0., -0.002, 0.00601202, 0.00199203,
+                                 0.00198807]),
+            'pnl': np.array([0., 0., -20000., 60000., 20000., 20000.]),
+            'capital_used': np.array([0., -120000., 0., 0., 0., 0.]),
+            'portfolio_value': np.array([10000000., 10000000., 9980000.,
+                                         10040000., 10060000., 10080000.])
+        }
+
+    def data_portal_mock(self, *args, **kwargs):
+        return self.data_portal
+
+    def asset_finder_mock(self, *args, **kwargs):
+        return self.data_portal.asset_finder
+
+    @classmethod
+    def validate_perf(cls, perf):
+        for stat, expected_output in cls.expected_daily.items():
+            np.testing.assert_array_almost_equal(
+                np.array(perf[stat]),
+                expected_output,
+                err_msg='daily ' + stat,
+            )
+
+    def test_run_no_parameter_and_no_bennchmark_set(self):
+        """
+        No benchmark parameter is provided and the benchmark is not set
+        in the algorithm's initialize function
+        Expected outcome : ValueError: Must specify either
+        benchmark_sid or benchmark_returns.
+        """
+        with mock.patch("zipline.utils.run_algo.DataPortal",
```

Comment:
We tend to try to avoid using `mock.patch` unless it's absolutely necessary, since patches tend to be fragile, and they usually mean you're depending on implementation details of the code being tested.

For this case, I think it might be easier for us to factor out the code that's responsible for constructing the benchmark from different possible inputs, and to test that code in isolation instead of having to run an entire algorithm to test the benchmark setup.